bundles/org.openhab.core.config.discovery.addon.mdns/src/main/java/org/openhab/core/config/discovery/addon/mdns/MdnsAddonFinder.java

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright (c) 2010-2023 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.core.config.discovery.addon.mdns;
+
+import static org.openhab.core.config.discovery.addon.AddonFinderConstants.SERVICE_NAME_MDNS;
+import static org.openhab.core.config.discovery.addon.AddonFinderConstants.SERVICE_TYPE_MDNS;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import javax.jmdns.ServiceEvent;
+import javax.jmdns.ServiceInfo;
+import javax.jmdns.ServiceListener;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.addon.AddonDiscoveryMethod;
+import org.openhab.core.addon.AddonInfo;
+import org.openhab.core.common.ThreadPoolManager;
+import org.openhab.core.config.discovery.addon.AddonFinder;
+import org.openhab.core.config.discovery.addon.BaseAddonFinder;
+import org.openhab.core.io.transport.mdns.MDNSClient;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Deactivate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is a {@link MdnsAddonFinder} for finding suggested Addons via MDNS.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution
+ * @author Mark Herwege - refactor to allow uninstall
+ */
+@NonNullByDefault
+@Component(service = AddonFinder.class, name = MdnsAddonFinder.SERVICE_NAME)
+public class MdnsAddonFinder extends BaseAddonFinder implements ServiceListener {
```

Comment:
In all other classes, we spell mDNS as MDNS -> may I ask you to refactor the names accordingly here, i.e. this one to `MDNSAddonFinder`?
(I agree that we are not consistent in our codebase - for UPnP, our classes are actually called Upnp... 🙄 )

Reply:
Ok.

But what about method names? In #3865 there is a getMdns.. and a setMdns.. mthod -- do you want me to change those too?